skglm/solvers/pn_solver.py

```diff
@@ -0,0 +1,295 @@
+import numpy as np
+from scipy.sparse import issparse
+from numba import njit
+
+from skglm.solvers.common import construct_grad, construct_grad_sparse
+
+
+def pn_solver(X, y, datafit, penalty, w_init=None, p0=10,
+              max_iter=20, max_epochs=1000, tol=1e-4, verbose=0):
+    """Run a Prox Newton solver.
```

Comment:
maybe add refs to Blitz/celer ? and add that it's combined with working sets (to explain `p0`)

Reply:
Should I put a link to Blitz/celer code?